graylog2-server/src/main/java/org/graylog2/web/customization/CustomizationConfig.java

```diff
@@ -19,18 +19,26 @@
 import jakarta.inject.Inject;
 
 import javax.annotation.Nullable;
+import java.util.Base64;
 import java.util.Optional;
 
+
 public class CustomizationConfig {
     private static final String DEFAULT_PRODUCT_NAME = "Graylog";
     private final Config config;
+    private final Base64.Decoder base64Decoder;
 
     @Inject
     public CustomizationConfig(@Nullable Config config) {
         this.config = Optional.ofNullable(config).orElse(Config.empty());
+        this.base64Decoder = Base64.getDecoder();
     }
 
     public String productName() {
         return config.productName().orElse(DEFAULT_PRODUCT_NAME);
     }
+
+    public Optional<byte[]> favicon() {
+        return config.favicon().map(base64Decoder::decode);
```

Comment:
Just a general thought - when I put wrong base64 string, I learn it relatively late, during first favicon() invocation, by exception thrown in the logs.

Do we consider to add some kind of validation for the Config class that checks it is ok, on startup (proper favicon, correctl helpUrl etc.)?

Reply:
Yes, why not. But maybe we do that when we open the feature for GA and expose it through the API.